lib/path.js

```diff
@@ -140,6 +140,24 @@ function normalizeStringPosix(path, allowAboveRoot) {
   return res;
 }
 
+function _format(sep, pathObject) {
+  if (pathObject === null || typeof pathObject !== 'object') {
+    throw new TypeError(
```

Comment:
Maybe keep the throws directly in ``path.format`` for better stack traces?
That would also avoid polymorphism in ``_format``